project_report/project_information.py

```diff
@@ -230,6 +236,12 @@ def get_project_library_workflows(self):
     def get_project_analysis_types(self):
         return self._aggregate_per_project(self.get_analysis_type_from_sample)
 
+    def has_rapid_samples(self):
+        return any(
+            query_dict(self.sample_info(s['sample_id']), 'info.Rapid Analysis') == 'Yes'
+            for s in self.sample_data_for_project
```

Comment:
Is this correct? It seems unfinished.

Reply:
Yup, [generator expression](https://docs.python.org/3/reference/expressions.html?highlight=generator#generator-expressions).